client/blocks/login/login-form.jsx

```diff
@@ -250,6 +260,10 @@ export class LoginForm extends Component {
 			return;
 		}
 
+		if ( isPasswordlessAccount( this.props.accountType ) ) {
+			this.props.sendMagicLoginLink?.();
+		}
```

Comment:
This pattern is pretty strange around signup logic, we get some info from `props` then we call some function from props, it seems like `props` already has all the info and tools to make the decision, why should the component be involved.